File: src/main/Java/dfs/boj1260.java
import java.io.*;
import java.util.*;
import java.util.stream.IntStream;

public class boj1260 {
    private static ArrayList<Integer>[] linked;
    private static boolean[] isVisited;
    private static StringBuilder builder = new StringBuilder();
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        StringTokenizer tokenizer = new StringTokenizer(reader.readLine());
        int n = Integer.parseInt(tokenizer.nextToken());
        int m = Integer.parseInt(tokenizer.nextToken());
        int v = Integer.parseInt(tokenizer.nextToken());
        linked = new ArrayList[n + 1];
        isVisited = new boolean[n + 1];
        IntStream.range(1, n + 1).forEach(index -> linked[index] = new ArrayList<>());
        while(m-- > 0) {
            tokenizer = new StringTokenizer(reader.readLine());
            int x = Integer.parseInt(tokenizer.nextToken());
            int y = Integer.parseInt(tokenizer.nextToken());
            linked[x].add(y);
            linked[y].add(x);
        }
        IntStream.range(1, n + 1).forEach(index -> Collections.sort(linked[index]));
        dfs(v);
        writer.append(builder);
        writer.newLine();
        IntStream.range(1, n + 1).forEach(index -> isVisited[index] = false);
        builder = new StringBuilder();
        bfs(v);
        writer.append(builder).close();
    }
    private static void dfs(int n) {
        if (isVisited[n]) return;
        isVisited[n] = true;
        builder.append(n).append(' ');
        linked[n].forEach(x -> dfs(x));
    }

    private static void bfs(int n) {
        Queue<Integer> q = new LinkedList<>();
        q.offer(n);
        builder.append(n).append(' ');
        isVisited[n] = true;
        while (!q.isEmpty()) {
            int c = q.poll();
            linked[c].forEach(it -> {
                if (!isVisited[it]) {
                    q.offer(it);
                    builder.append(it).append(' ');
                    isVisited[it] = true;
                }
            });
        }
    }
}

File: src/main/kotlin/dfs/boj1260.kt

import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.util.*
import java.util.stream.IntStream

private var linked = arrayOf(ArrayList<Int>())
private var isVisited = booleanArrayOf()
private val queue = LinkedList<Int>()
fun main() {
    val r = BufferedReader(InputStreamReader(System.`in`))
    val w = BufferedWriter(OutputStreamWriter(System.out))
    val (n, m, v) = r.readLine().split(" ").map { it.toInt() }
    linked = Array(n + 1) { arrayListOf() }
    IntStream.range(1, n + 1).forEach { linked[it] = ArrayList() }
    isVisited = BooleanArray(n + 1)
    repeat (m) {
        val (x, y) = r.readLine().split(" ").map { it.toInt() }
        linked[x].add(y)
        linked[y].add(x)
    }
    linked.forEach { it.sort() }
//    if (!isVisited[v]) dfs(v)
//    IntStream.range(1, n + 1).forEach { if (!isVisited[it]) dfs(it) }
    dfs(v)
    IntStream.range(1, n + 1).forEach { isVisited[it] = false }
    queue.forEach { w.append("$it ") }
    w.newLine()
    queue.clear()
    bfs(v)
    queue.forEach { w.append("$it ") }
    w.close()
}

//private fun dfs(n: Int) {
//    if (isVisited[n]) return
//    isVisited[n] = true
//    queue.offer(n)
//    linked[n].forEach { dfs(it) }
//}

private fun dfs(n: Int) {
    if (isVisited[n]) return
    queue.offer(n)
    isVisited[n] = true
    linked[n].forEach { dfs(it) }
}
//private fun bfs(n: Int) {
//    isVisited[n] = true
//    linked[n].forEach { if(!isVisited[it]) queue.offer(it) }
//    linked[n].forEach { if (!isVisited[it]) bfs(it) }
//}

private fun bfs(n: Int) {
    val q: Queue<Int> = LinkedList()
    q.offer(n)
    queue.offer(n)
    isVisited[n] = true
    while (q.isNotEmpty()) {
        val c = q.poll()
        linked[c].forEach {
            if (!isVisited[it]) {
                q.offer(it)
                queue.offer(it)
                isVisited[it] = true
            }
        }
    }
}